pkg/healthcheck/healthcheck.go

```diff
@@ -1353,6 +1370,27 @@ func (hc *HealthChecker) checkCanPerformAction(verb, namespace, group, version,
 	)
 }
 
+func (hc *HealthChecker) checkHAMetadataPresentOnKubeSystemNamespace() error {
+	if hc.isHA() {
```

Comment:
It is probably out of scope for this PR, but should we move the HA part to the check definition itself?

Reply:
This will make testing the case when we do not have ha enabled a bit cumbersome I think